### PR TITLE
[WIP]Feat: add EIP-1559 transaction type

### PR DESCRIPTION
**Pull request by haythemsellami**

### CODE REVIEW

File: src/Transactions.sol [hunk 1]
```diff
@@ -38,7 +53,26 @@ library Transactions {
         return RLPWriter.writeList(items);
     }
 
-    function decodeRLP(bytes memory rlp) internal pure returns (Legacy memory) {
+    function encodeRLP(EIP1559 memory txStruct) internal pure returns (bytes memory) {
+        bytes[] memory items = new bytes[](12);
```
Comment (ferranbt) [inline src/Transactions.sol:57]: Is this encoding correct? `EIP-1559` transactions are encoded with [`EIP-2718`](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2718.md) format, so it needs to have the transaction type prefixed. You can find a code reference [here](https://github.com/Lilyjjo/tldr-research/blob/main/poc_only_suave/suave-contracts/src/SuaveSigner.sol#L128).

File: src/Transactions.sol [hunk 2]
```diff
@@ -22,7 +22,22 @@ library Transactions {
...
+        uint64 nonce;
+        bytes data;
+        uint64 chainId;
+        bytes[] accessList;
+        bytes r;
+        bytes s;
+        bytes v;
+    }
+
+    function encodeLegacyRLP(Legacy memory txStruct) internal pure returns (bytes memory) {
```
Comment (ferranbt) [inline src/Transactions.sol:40]: Can you keep the same name here? I imagine having the same `encodeRLP` name overloaded for both transactions.

**Review by ferranbt — CHANGES_REQUESTED** [review]
Hey! Thank you for your contribution! That was fast!

I left a couple of comments. Could you also include a test to validate the encoding/decoding?

**Comment by ferranbt** [comment]
I made some changes to the encoding of legacy transactions. You'll need to pull the new changes.

**Comment by haythemsellami** [comment]
> I made some changes to the encoding of legacy transactions. You'll need to pull the new changes.

Cool will do.

File: src/Transactions.sol (hunk 1 above)
Comment (haythemsellami) [reply src/Transactions.sol]: I just fixed this, lmk if it is still not correct.

**Comment by ferranbt** [comment]
Hey @haythemsellami. Do you need any help on the PR?

**Comment by haythemsellami** [comment]
> Hey @haythemsellami. Do you need any help on the PR?

Trying to finish the test and make it working, but I may have a bug in the encoding implementation as the output is not matching(maybe the access list encoding is wrong?)

**Comment by haythemsellami** [comment]
Closing this in favour of #36.

**Pull request closed without merging**